content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -175,6 +175,12 @@ the possible following fields:
 
 `dvc.yaml` files also support `# comments`.
 
+We have published the
+[JSON Schema](https://github.com/iterative/pipeline-schema) for `dvc.yaml` files
+that helps code editors and IDEs to provide auto-completion and
+validation/linting and to verify proper structure of `dvc.yaml` locally and/or
+in CI.
```

Comment:
Questions that may need explaining/ clarifying here:
- What's are the main goals for publishing the JSON schema? Command line validation/linting tools tools, CI checks, IDE auto-completion?
- Why is it a JSON schema if the file format is YAML?

Also, should we rename the repo `dvcyaml-schema` instead of `pipeline-schema`? We want to avoid synonyms so we never use the term "pipelines file", it's always dvc.yaml.

Reply:
See also https://github.com/iterative/pipeline-schema/issues/4 🙂

Reply:
> What are the main goals for publishing the JSON schema?

It's the autocompletion and validation/linting inside text editors for now. 

> Why is it a JSON schema if the file format is YAML?

Most of the editors require extensions that use Redhat's YAML language server, which uses JSON schema to validate, because:
1. JSON is a subset of YAML.
2. It's just a schema, so if any format is largely compatible with JSON, it can be described through that schema (though, it might be a bit restrictive).
3. Describing YAML with a different set of the schema (let's say, YAML Schema) is a ton of work, given the fact that JSON Schema is still not standardized.

Reply:
I agree with @jorgeorpinel , JSON will be confusing. Docs are not about being mathematically correct (especially docs outside command reference). Their goal is to provide useful information to users in the most accessible way.

We need to simplify the paragraph. It can be something like:

💡 There is a `dvc.yaml` [schema](link) available that could be installed into editors like [VSCode](link) or [PyCharm](link) to enable automatic syntax checks and autocompletion.